### PR TITLE
Contacts

### PR DESCRIPTION
**Pull request by grishmashrestha**

### CODE REVIEW

File: app/components/AddToContactButton/AddToContactButton.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        alert('Cant add to contacts. Please give permission.')
+      }
+    })
+  }
+
+  _addEmployeeToContact() {
+    alert('Authorized')
+    Contacts.getAll((err, contacts) => {
+      console.log(contacts[0])
+      // check for old contact
```
Comment (leapfrog-saugat) [inline app/components/AddToContactButton/AddToContactButton.js:32]: Add TODO for tasks to be done.

File: app/components/AddToContactButton/AddToContactButton.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        number: employee.contact.mobilePhone,
+      }, {
+        label: "home",
+        number: employee.contact.homePhone,
+      }]
+    };
+
+    Contacts.addContact(newPerson, (err) => {
+      if (err) {
+        alert('Contact could not be created', err);
```
Comment (leapfrog-saugat) [inline app/components/AddToContactButton/AddToContactButton.js:62]: Be consistent with the use of ;

File: app/components/AddToContactButton/index.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import AddToContactButton from './AddToContactButton';
+
+export default AddToContactButton;
```
Comment (leapfrog-saugat) [inline app/components/AddToContactButton/index.js:3]: Discuss the convention for an extra line space at the end of the file.

File: app/routes/ApplicationShell/ApplicationShell.ios.js [hunk 4]
```diff
@@ -1,28 +1,90 @@
...
-        super(props);
-    }
+		super(props);
+		this._renderNavigationView = this._renderNavigationView.bind(this);
+		this._renderScene = this._renderScene.bind(this);
+		this._handleBackAction = this._handleBackAction.bind(this);
+	}
+
+	componentDidMount() {
+		BackAndroid.addEventListener('hardwareBackPress', this._handleBackAction)
```
Comment (leapfrog-saugat) [inline app/routes/ApplicationShell/ApplicationShell.ios.js:28]: 2 space

File: app/routes/ApplicationShell/ApplicationShell.ios.js [hunk 5]
```diff
@@ -1,28 +1,90 @@
...
+		return (
+			<View style={{flex: 1, backgroundColor: '#fff'}}>
+	<TouchableHighlight  onPress={() => this._navigatePage({key: 'home'})}>
+	<Text>Home</Text>
+		</TouchableHighlight>
+		<TouchableHighlight  onPress={() => this._navigatePage({key: 'favourites'})}>
+	<Text>Favourite</Text>
+		</TouchableHighlight>
+		</View>
+	);
```
Comment (leapfrog-saugat) [inline app/routes/ApplicationShell/ApplicationShell.ios.js:56]: Check the indentation.

File: app/routes/ApplicationShell/ApplicationShell.ios.js [hunk 6]
```diff
@@ -1,28 +1,90 @@
...
+	}
+
+	_renderScene(props) {
+		const { route } = props.scene;
+
+		switch(route.key){
+
+			case 'home':
+			console.log("****ROUTE****", route.key)
```
Comment (leapfrog-saugat) [inline app/routes/ApplicationShell/ApplicationShell.ios.js:65]: Remove unnecessary logs

File: app/routes/ApplicationShell/ApplicationShell.ios.js [hunk 7]
```diff
@@ -1,28 +1,90 @@
...
+	<Text>Favourite</Text>
+		</TouchableHighlight>
+		</View>
+	);
+	}
+
+	_renderScene(props) {
+		const { route } = props.scene;
+
+		switch(route.key){
```
Comment (leapfrog-saugat) [inline app/routes/ApplicationShell/ApplicationShell.ios.js:62]: be consistent with the spacing before  { .

File: app/routes/EmployeeDetails/styles.js [hunk 8]
```diff
@@ -61,24 +66,25 @@ const styles = StyleSheet.create({
...
   contactInfo: {
-    fontSize: 12
+    fontSize: 14,
+    lineHeight: 16,
   },
   contactInfoMore: {
-    fontSize: 11
+    fontSize: 10,
+    lineHeight: 12,
+    color: '#9B9D9C'
```
Comment (leapfrog-saugat) [inline app/routes/EmployeeDetails/styles.js:87]: We might need to separate out colors in a separate color constant file

File: app/routes/EmployeeDetails/index.js [hunk 9]
```diff
@@ -1,3 +1,3 @@
-import EmployeeDetails from './EmployeeDetails';
+import EmployeeDetails from './EmployeeDetails.android';
```
Comment (leapfrog-saugat) [inline app/routes/EmployeeDetails/index.js]: Remove .android as well please so as to make it work on iPhones as well :)

File: app/utils/EmployeeUtils.js [hunk 10]
```diff
@@ -1,4 +1,11 @@
 export function getEmployeeFullName({firstName, middleName, lastName}){
-	let mName = ( middleName === '' || middleName === null ) ? '' : middleName + ' ';
-	return `${firstName} ${mName}${lastName}`;
-}
\ No newline at end of file
+	// if ((firstName == null) || (middleName == null) || (lastName == null)) {
```
Comment (leapfrog-saugat) [inline app/utils/EmployeeUtils.js]: Add TODO

File: app/components/AddToContactButton/index.js (hunk 3 above)
Comment (grishmashrestha) [reply app/components/AddToContactButton/index.js:3]: :+1: okay. We will discuss this on our next meeting.

File: app/routes/EmployeeDetails/styles.js (hunk 8 above)
Comment (grishmashrestha) [reply app/routes/EmployeeDetails/styles.js:87]: Sound good. Duly noted.

**Pull request merged**